### ISSUE TITLE
Deliver not setting price tier for new app submissions

### ISSUE THREAD

**Issue by andrewdieken** [fastlane/fastlane#21370]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I'm attempting to release a new app and am running into the error described https://github.com/fastlane/fastlane/issues/21125#issuecomment-1474628335. The suggested fix resolves this error but I am now unable to add the app for review because the price tier is __not__ being set forcing me to complete the Pricing and Availability section manually within App Store Connect.

![fastlane1](https://github.com/fastlane/fastlane/assets/27448527/4d3b49aa-9169-4470-892e-2fc8a4219bd3)

##### Command executed
`fastlane deliver --overwrite_screenshots true --force true`

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> [INFO - > fastlane deliver --overwrite_screenshots true --force true
DEBUG - [10:55:20]: [33mGet started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile[0m
DEBUG - [10:55:21]: [32mSuccessfully loaded '/redacted_file_path/Deliverfile' 📄[0m
DEBUG - 
DEBUG - +------------------------+--------------------------------------+
DEBUG - |             Detected Values from './Deliverfile'              |
DEBUG - +------------------------+--------------------------------------+
DEBUG - | app_identifier         | redacted                             |
DEBUG - | ipa                    | /redacted-Distribution.ipa            |
DEBUG - | app_version            | 1.0.0                                |
DEBUG - | submit_for_review      | false                                |
DEBUG - | screenshots_path       | redacted                             |
DEBUG - | metadata_path          | redacted                             |
DEBUG - | price_tier             | 0                                    |
DEBUG - | app_review_information | {:first_name=>"redacted",            |
DEBUG - |                        | :last_name=>"redacted",              |
DEBUG - |                        | :phone_number=>"redacted",           |
DEBUG - |                        | :email_address=>"redacted",          |
DEBUG - |                        | :demo_user=>"redacted",              |
DEBUG - |                        |  :demo_password=>"redacted",         |
DEBUG - |                        | :notes=>"\n\nPlease be sure to       |
DEBUG - |                        | delete any testing comments made in  |
DEBUG - |                        | the guide."}                         |
DEBUG - | submission_information | {:add_id_info_uses_idfa=>false,      |
DEBUG - |                        | :export_compliance_encryption_updat  |
DEBUG - |                        | ed=>false,                           |
DEBUG - |                        | :export_compliance_uses_encryption=  |
DEBUG - |                        | >false,                              |
DEBUG - |                        | :content_rights_contains_third_part  |
DEBUG - |                        | y_content=>false}                    |
DEBUG - | automatic_release      | true                                 |
DEBUG - | app_rating_config_path | redacted                             |
DEBUG - | app_icon               | redacted                             |
DEBUG - +------------------------+--------------------------------------+
DEBUG - 
DEBUG - [10:55:21]: Login to App Store Connect (redacted)
DEBUG - [10:55:24]: Login successful
DEBUG - 
DEBUG - +--------------------------------------+--------------------------------------+
DEBUG - |                           [32mdeliver 2.212.2 Summary[0m                           |
DEBUG - +--------------------------------------+--------------------------------------+
DEBUG - | overwrite_screenshots                | true                                 |
DEBUG - | force                                | true                                 |
DEBUG - | username                             | redacted                             |
DEBUG - | app_identifier                       | redacted                             |
DEBUG - | app_version                          | 1.0.0                                |
DEBUG - | ipa                                  | redacted                             |
DEBUG - | platform                             | ios                                  |
DEBUG - | edit_live                            | false                                |
DEBUG - | use_live_version                     | false                                |
DEBUG - | metadata_path                        | redacted                             |
DEBUG - | screenshots_path                     | redacted                             |
DEBUG - | skip_binary_upload                   | false                                |
DEBUG - | skip_screenshots                     | false                                |
DEBUG - | skip_metadata                        | false                                |
DEBUG - | skip_app_version_update              | false                                |
DEBUG - | sync_screenshots                     | false                                |
DEBUG - | submit_for_review                    | false                                |
DEBUG - | verify_only                          | false                                |
DEBUG - | reject_if_possible                   | false                                |
DEBUG - | automatic_release                    | true                                 |
DEBUG - | phased_release                       | false                                |
DEBUG - | reset_ratings                        | false                                |
DEBUG - | price_tier                           | 0                                    |
DEBUG - | app_rating_config_path               | redacted                             |
DEBUG - | submission_information.add_id_info_  | false                                |
DEBUG - | uses_idfa                            |                                      |
DEBUG - | submission_information.export_compl  | false                                |
DEBUG - | iance_encryption_updated             |                                      |
DEBUG - | submission_information.export_compl  | false                                |
DEBUG - | iance_uses_encryption                |                                      |
DEBUG - | submission_information.content_righ  | false                                |
DEBUG - | ts_contains_third_party_content      |                                      |
DEBUG - | run_precheck_before_submit           | true                                 |
DEBUG - | precheck_default_rule_level          | warn                                 |
DEBUG - | app_icon                             | redacted                             |
DEBUG - | app_review_information.first_name    | redacted                             |
DEBUG - | app_review_information.last_name     | redacted                             |
DEBUG - | app_review_information.phone_number  | redacted                             |
DEBUG - | app_review_information.email_addres  | redacted                             |
DEBUG - | s                                    |                                      |
DEBUG - | app_review_information.demo_user     | redacted                             |
DEBUG - | app_review_information.demo_passwor  | ********                             |
DEBUG - | d                                    |                                      |
DEBUG - | app_review_information.notes         |   Please be sure to delete any       |
DEBUG - |                                      | testing comments made in the guide.  |
DEBUG - | ignore_language_directory_validatio  | false                                |
DEBUG - | n                                    |                                      |
DEBUG - | precheck_include_in_app_purchases    | true                                 |
DEBUG - +--------------------------------------+--------------------------------------+
DEBUG - 
DEBUG - [10:55:25]: Making sure the latest version on App Store Connect matches '1.0.0'...
DEBUG - [10:55:25]: [32mSuccessfully set the version to '1.0.0'[0m
DEBUG - [10:55:25]: Loading '/redacted/submission_files/metadata/en-US/description.txt'...
DEBUG - [10:55:25]: Loading '/redacted/submission_files/metadata/en-US/keywords.txt'...
DEBUG - [10:55:25]: Loading '/redacted/submission_files/metadata/en-US/release_notes.txt'...
DEBUG - [10:55:25]: Loading '/redacted/submission_files/metadata/en-US/support_url.txt'...
DEBUG - [10:55:25]: Loading '/redacted/submission_files/metadata/en-US/marketing_url.txt'...
DEBUG - [10:55:25]: Loading '/redacted/submission_files/metadata/en-US/name.txt'...
DEBUG - [10:55:25]: Loading '/redacted/submission_files/metadata/en-US/privacy_url.txt'...
DEBUG - [10:55:25]: Loading '/redacted/submission_files/metadata/copyright.txt'...
DEBUG - [10:55:25]: Loading '/redacted/submission_files/metadata/primary_category.txt'...
DEBUG - [10:55:28]: [33mWill begin uploading metadata for '1.0.0' on App Store Connect[0m
DEBUG - [10:55:28]: [31mSkipping 'release_notes'... this is the first version of the app[0m
DEBUG - [10:55:28]: Uploading metadata to App Store Connect for version
DEBUG - [10:55:31]: Uploading metadata to App Store Connect for localized version 'en-US'
DEBUG - [10:55:32]: Uploading metadata to App Store Connect for localized info 'en-US'
DEBUG - [10:55:33]: Uploading app review information to App Store Connect
DEBUG - [10:55:33]: [31mError fetching app store review detail - No data[0m
DEBUG - [10:55:34]: Setting the app's age rating...
DEBUG - [10:55:35]: [33mWill begin uploading snapshots for '1.0.0' on App Store Connect[0m
DEBUG - [10:55:35]: Starting with the upload of screenshots...
DEBUG - [10:55:35]: [33mNumber of screenshots not deleted: 0[0m
DEBUG - [10:55:35]: Successfully deleted all screenshots
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/1apple_screenshot_65.jpeg'... (1.492475 secs)
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/3apple_screenshot_55.jpeg'... (1.553224 secs)
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/4apple_screenshot_55.jpeg'... (1.555817 secs)
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/1apple_screenshot_55.jpeg'... (1.563293 secs)
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/1apple_screenshot_ipadPro129.jpeg'... (1.681382 secs)
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/2apple_screenshot_65.jpeg'... (1.836794 secs)
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/2apple_screenshot_55.jpeg'... (1.837931 secs)
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/2apple_screenshot_ipadPro129.jpeg'... (1.837845 secs)
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/3apple_screenshot_65.jpeg'... (1.858713 secs)
DEBUG - [10:55:38]: Uploaded '/redacted/submission_files/screenshots/en-US/4apple_screenshot_65.jpeg'... (1.874443 secs)
DEBUG - [10:55:39]: Uploaded '/redacted/submission_files/screenshots/en-US/1apple_screenshot_ipadpro.jpeg'... (1.38737 secs)
DEBUG - [10:55:39]: Uploaded '/redacted/submission_files/screenshots/en-US/2apple_screenshot_ipadpro.jpeg'... (1.396996 secs)
DEBUG - [10:55:39]: Uploaded '/redacted/submission_files/screenshots/en-US/3apple_screenshot_ipadPro129.jpeg'... (1.516935 secs)
DEBUG - [10:55:39]: Uploaded '/redacted/submission_files/screenshots/en-US/3apple_screenshot_ipadpro.jpeg'... (1.451966 secs)
DEBUG - [10:55:51]: Successfully uploaded all screenshots
DEBUG - [10:55:51]: [32mSuccessfully uploaded screenshots to App Store Connect[0m
DEBUG - [10:55:51]: App has no prices yet... No changes required.
DEBUG - [10:55:51]: Uploading binary to App Store Connect
DEBUG - [10:55:51]: Fetching password for transporter from environment variable named `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD`
DEBUG - [10:55:55]: Fetching password for transporter from environment variable named `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD`
DEBUG - [10:55:55]: Going to upload updated app to App Store Connect
DEBUG - [10:55:55]: [32mThis might take a few minutes. Please don't interrupt the script.[0m
DEBUG - [10:57:22]: [32m--------------------------------------------------------------------[0m
DEBUG - [10:57:22]: [32mSuccessfully uploaded package to App Store Connect. It might take a few minutes until it's visible online.[0m
DEBUG - [10:57:22]: [32m--------------------------------------------------------------------[0m
DEBUG - [10:57:22]: [32mFinished the upload to App Store Connect[0m
DEBUG - [10:57:22]: Running precheck before submitting to review, if you'd like to disable this check you can set run_precheck_before_submit to false
DEBUG - [10:57:22]: Running precheck 👮‍♀️ 👮
DEBUG - 
DEBUG - +--------------------------+---------------------------+
DEBUG - |             [32mSummary for precheck 2.212.2[0m             |
DEBUG - +--------------------------+---------------------------+
DEBUG - | default_rule_level       | warn                      |
DEBUG - | include_in_app_purchases | true                      |
DEBUG - | app_identifier           | redacted                  |
DEBUG - | username                 | redacted                  |
DEBUG - | platform                 | ios                       |
DEBUG - | use_live                 | false                     |
DEBUG - +--------------------------+---------------------------+
DEBUG - 
DEBUG - [10:57:22]: Checking app for precheck rule violations
DEBUG - [10:57:25]: Reading in-app purchases. If you have a lot, this might take a while
DEBUG - [10:57:25]: You can disable IAP checking by setting the `include_in_app_purchases` flag to `false`
DEBUG - [10:57:25]: Done reading in-app purchases
DEBUG - [10:57:25]: ✅  Passed: No negative  sentiment
DEBUG - [10:57:25]: ✅  Passed: No placeholder text
DEBUG - [10:57:25]: ✅  Passed: No mentioning  competitors
DEBUG - [10:57:25]: ✅  Passed: No future functionality promises
DEBUG - [10:57:25]: ✅  Passed: No words indicating test content
DEBUG - [10:57:25]: ✅  Passed: No curse words
DEBUG - [10:57:25]: ✅  Passed: No words indicating your IAP is free
DEBUG - [10:57:25]: ✅  Passed: Incorrect, or missing copyright date
DEBUG - [10:57:26]: [33m😵  Failed: No broken urls-> unreachable URLs in app metadata[0m
DEBUG - +----------------------+------------------------------+
DEBUG - |                 [33mPotential problems[0m                  |
DEBUG - +----------------------+------------------------------+
DEBUG - | Field                | Failure reason               |
DEBUG - +----------------------+------------------------------+
DEBUG - | support URL: (en-US) | [33mhttp://support.guidebook.com[0m |
DEBUG - +----------------------+------------------------------+
DEBUG - [10:57:26]: [33m[33mprecheck 👮‍♀️ 👮  found one or more potential metadata problems, but this won't prevent fastlane from completing 👍[] </pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 12.6.6                                      |
| Ruby                        | 2.7.3                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.37.1 (Apple Git-137.1)        |
| Installation Source         | ~/.rvm/gems/ruby-2.7.3/bin/fastlane         |
| Host                        | macOS 12.6.6 (21G646)                       |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.7.3/lib                |
| OpenSSL Version             | OpenSSL 1.1.1q  5 Jul 2022                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 14.2                                        |
| Swift Version               | 5.7.2                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.213.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.4.0        |
| executable-hooks                | 1.6.1        |
| bundler-unload                  | 1.0.2        |
| rubygems-bundler                | 1.4.5        |
| bundler                         | 2.1.4        |
| atomos                          | 0.1.3        |
| CFPropertyList                  | 3.0.5        |
| claide                          | 1.1.0        |
| colored2                        | 3.1.2        |
| nanaimo                         | 0.3.0        |
| rexml                           | 3.2.5        |
| xcodeproj                       | 1.22.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.8.0        |
| terminal-table                  | 1.8.0        |
| plist                           | 3.6.0        |
| public_suffix                   | 4.0.7        |
| addressable                     | 2.8.0        |
| multipart-post                  | 2.0.0        |
| word_wrap                       | 1.0.0        |
| optparse                        | 0.1.1        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| artifactory                     | 3.0.15       |
| babosa                          | 1.0.4        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| excon                           | 0.92.3       |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| faraday-multipart               | 1.0.4        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| faraday-retry                   | 1.0.3        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.10.0       |
| unf_ext                         | 0.0.8.2      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| http-cookie                     | 1.0.5        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.2.0        |
| fastimage                       | 2.2.6        |
| gh_inspector                    | 1.1.3        |
| json                            | 2.3.0        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.3        |
| xcpretty-travis-formatter       | 1.0.1        |
| dotenv                          | 2.7.6        |
| simctl                          | 1.6.8        |
| jwt                             | 2.4.1        |
| uber                            | 0.1.0        |
| declarative                     | 0.0.20       |
| trailblazer-option              | 0.1.2        |
| representable                   | 3.2.0        |
| retriable                       | 3.1.2        |
| mini_mime                       | 1.1.2        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| signet                          | 0.17.0       |
| googleauth                      | 1.2.0        |
| httpclient                      | 2.8.3        |
| webrick                         | 1.6.1        |
| google-apis-core                | 0.7.0        |
| google-apis-playcustomapp_v1    | 0.10.0       |
| google-apis-androidpublisher_v3 | 0.25.0       |
| google-cloud-env                | 1.6.0        |
| google-cloud-errors             | 1.2.0        |
| google-cloud-core               | 1.6.0        |
| google-apis-iamcredentials_v1   | 0.13.0       |
| google-apis-storage_v1          | 0.18.0       |
| rake                            | 13.0.1       |
| digest-crc                      | 0.6.4        |
| google-cloud-storage            | 1.37.0       |
| emoji_regex                     | 3.2.3        |
| jmespath                        | 1.6.1        |
| aws-partitions                  | 1.605.0      |
| aws-eventstream                 | 1.2.0        |
| aws-sigv4                       | 1.5.0        |
| aws-sdk-core                    | 3.131.2      |
| aws-sdk-kms                     | 1.57.0       |
| aws-sdk-s3                      | 1.114.0      |
| uri                             | 0.10.0       |
| forwardable                     | 1.3.1        |
| logger                          | 1.4.2        |
| cgi                             | 0.1.0        |
| date                            | 3.0.0        |
| timeout                         | 0.1.0        |
| stringio                        | 0.1.0        |
| ipaddr                          | 1.2.2        |
| openssl                         | 2.1.2        |
| zlib                            | 1.1.0        |
| mutex_m                         | 0.1.0        |
| ostruct                         | 0.2.0        |
| strscan                         | 1.0.3        |
| io-console                      | 0.5.6        |
| delegate                        | 0.1.0        |
| fileutils                       | 1.4.1        |
| singleton                       | 0.1.0        |
| open3                           | 0.1.0        |
| yaml                            | 0.1.0        |
| psych                           | 3.1.0        |
</details>

*generated on:* **2023-07-03**
</details>

**Comment by pedro380085** [comment]
I have the same error as well.

**Comment by Glaubenio** [comment]
Same problem here

**Comment by JoaoVytor** [comment]
The error is happening in the project I'm working too

**Comment by rafael-sevla** [comment]
Error occurring here too

**Comment by luanalisboamartins** [comment]
here also the error occurred

**Comment by andrewdieken** [comment]
This continues to be an issue for me

**Comment by bkrijgsman** [comment]
Also have the issue overhere!

**Comment by assembleMHN** [comment]
same, issue still persists

**Comment by danielftz** [comment]
Same problem still

**Comment by kirakoki** [comment]
is this being worked on ? 

also would be nice to have availability sorted as well.

**Comment by lakshydeep-14** [comment]
Facing same issue.
Any solution?

**Comment by ersel** [comment]
is this even possible to automate?

https://forums.developer.apple.com/forums/thread/722733?page=2

I found this rabbit hole on apple developer forums but looks like the relevant endpoints are poorly documented.

**Comment by france-workmail** [comment]
Any new updates on this for `deliver` action ? setting `price_tier` seems to pass the checks but it fails on the actual submission making me set the prices manually on appstore connect.

**Comment by pedro380085** [comment]
There is some work being done on a new PR right here https://github.com/fastlane/fastlane/issues/21968#issuecomment-2052408679

**Comment by france-workmail** [comment]
> There is some work being done on a new PR right here [#21968 (comment)](https://github.com/fastlane/fastlane/issues/21968#issuecomment-2052408679)

Thanks @pedro380085 ! Hopefully we get that soon.

**Comment by assembleMHN** [comment]
Still having this issue for over a year now :(

**Comment by ScottBishop** [comment]
Same issue here

**Comment by moazelsawaf** [comment]
Same issue

**Comment by assembleMHN** [comment]
Still this issue is here. Issue happens when you create a new app with no pricing and try to upload submit. We have it automated so app is created, metadata is created and uploaded and app is build and uploaded. But it breaks when fastlane tries to submit because Fastlane no longer set pricing for the app. Please fix this, it breaks our entire fastlane pipeline for new apps which we use several times a week for multiple apps. Somebody please look into this.

**Comment by france-workmail** [comment]
> Still this issue is here. Issue happens when you create a new app with no pricing and try to upload submit. We have it automated so app is created, metadata is created and uploaded and app is build and uploaded. But it breaks when fastlane tries to submit because Fastlane no longer set pricing for the app. Please fix this, it breaks our entire fastlane pipeline for new apps which we use several times a week for multiple apps. Somebody please look into this.

Been waiting for an update on this one and was hopeful with the last update since the PR mentioned was there, but unfortunately it wasn't for the one that sets prices and availabilities.

**Comment by cytryn** [comment]
Same issue, also waiting on a potential fix.

**Comment by france-workmail** [comment]
You can combine it with your existing workflows. For example, after you run the produce(...) action lane you can get the app id and then just pass it to setAppPrice lane:

```
produce(....)
app_identifier = lane_context[SharedValues::PRODUCE_APPLE_ID]
setAppPrice(app_id: app_identifier)
```

**Comment by baskrijgsman** [comment]
@france-workmail wow sounds promising. Will be testing this out soon :-)

**Comment by DouglasSouzaGomes** [comment]
@france-workmail thank you, this works nicely!

**Comment by france-workmail** [comment]
@DouglasSouzaGomes No problem. Though I'd like to note that I am still missing a way to set the availability of the app to pair with the pricing.
